src/config.rs

```diff
@@ -148,29 +150,37 @@ impl Config {
     }
 
     /// Retrieves all configured auth/delete tokens.
-    pub fn get_tokens(&self, token_type: TokenType) -> Option<Vec<String>> {
-        let mut tokens = match token_type {
+    pub fn get_tokens(&self, token_type: TokenType) -> Option<HashSet<String>> {
+        let tokens = match token_type {
             TokenType::Auth => {
-                let mut tokens = self.server.auth_tokens.clone().unwrap_or_default();
+                let mut tokens: HashSet<_> = self.server.auth_tokens.clone().unwrap_or_default();
+
                 #[allow(deprecated)]
                 if let Some(token) = &self.server.auth_token {
-                    tokens.insert(0, token.to_string());
+                    tokens.insert(token.to_string());
                 }
-                if let Ok(env_token) = env::var(AUTH_TOKEN_ENV) {
-                    tokens.insert(0, env_token);
+                match env::var(AUTH_TOKEN_ENV) {
+                    Ok(env_token) if !env_token.is_empty() => {
+                        tokens.insert(env_token);
+                    }
+                    _ => {}
                 }
                 tokens
             }
             TokenType::Delete => {
-                let mut tokens = self.server.delete_tokens.clone().unwrap_or_default();
-                if let Ok(env_token) = env::var(DELETE_TOKEN_ENV) {
-                    tokens.insert(0, env_token);
+                let mut tokens: HashSet<_> = self.server.delete_tokens.clone().unwrap_or_default();
+
+                match env::var(DELETE_TOKEN_ENV) {
+                    Ok(env_token) if !env_token.is_empty() => {
+                        tokens.insert(env_token);
+                    }
+                    _ => {}
                 }
                 tokens
             }
         };
-        tokens.retain(|v| !v.is_empty());
-        (!tokens.is_empty()).then_some(tokens)
+
+        Some(tokens).filter(|v| !v.is_empty())
```

Comment:
Can you add an unit test to verify this behavior?

Reply:
Sure!
Btw, we do filter out only empty string (`""` ), don't we want to filter whitespace strings and etc? 🤔

Reply:
I changed to filter blank strings as well, it looks more secure to me

Reply:
Yeah, I think not allowing whitespace strings makes perfect sense. Thanks for adding the tests!